ntbs-service/DataMigration/NotificationImportService.cs

```diff
@@ -0,0 +1,126 @@
+
+            LogInformation($"{notifications.Count} notifications included linked ones found to import");
+            var notificationsToSave = new List<Notification>();
+            foreach (var notification in notifications)
+            {
+                var linkedNotificationId = notification.LTBRID ?? notification.ETSID;  
+                LogInformation($"Validating notification with Id={linkedNotificationId}");
+
+                var validationErrors = GetValidationErrors(notification);
+                
+                if (validationErrors.Count() == 0)
+                {
+                    notificationsToSave.Add(notification);
+                    LogInformation($"No validation errors found");
+                }
+                else
+                {
+                    LogInformation($"{validationErrors.Count()} validation errors found:");
+                    foreach (var validationError in validationErrors)
+                    {
+                        LogInformation(validationError.ErrorMessage);
+                    }
+                }
+            }
+            LogInformation($"Importing {notificationsToSave.Count()} valid notifications");
+            var savedNotificaitons = await AddLinkedNotificationsAsync(notificationsToSave);
+
+
+            LogInformation($"Finished importing notification with Id={notificationId} and linked notifications");
+            return savedNotificaitons;
+        }
+
+        private IEnumerable<ValidationResult> GetValidationErrors(Notification notification) 
+        {
+            var validationsResults = new List<ValidationResult>();
+
+            notification.ShouldValidateFull = true;  
+            notification.PatientDetails.ShouldValidateFull = true;  
+            notification.ClinicalDetails.ShouldValidateFull = true;  
+            notification.TravelDetails.ShouldValidateFull = true;  
+            notification.VisitorDetails.ShouldValidateFull = true;  
+
+            validationsResults.AddRange(ValidateObject(notification));
+            validationsResults.AddRange(ValidateObject(notification.PatientDetails));
+            validationsResults.AddRange(ValidateObject(notification.ClinicalDetails));
+            validationsResults.AddRange(ValidateObject(notification.TravelDetails));
+            validationsResults.AddRange(ValidateObject(notification.VisitorDetails));
+
+            return validationsResults;
+        }
+
+        private IEnumerable<ValidationResult> ValidateObject<T>(T objectToValidate)
+        {
+            var validationContext = new ValidationContext(objectToValidate);
+            var validationsResults = new List<ValidationResult>();  
+
+            Validator.TryValidateObject(objectToValidate, validationContext, validationsResults, true);
+
+            return validationsResults;
+        }
+
+        private void LogInformation(string message)
+        {
+            Log.Information($"NOTIFICATION IMPORT: {message}");

```

Comment:
Would it also be worth adding some sort of run identifier? Perhaps the request id, or similar?